src/pages/CreateFeedback.tsx

```diff
@@ -58,8 +60,19 @@ export const CreateFeedback = (props: IProps) => {
       comments: [],
     };
 
-    productRequests.push(newFeedback);
-    navigate('/');
+    if (title.length === 0 && feedback.length === 0) {
```

Comment:
Do we need this? Why can't we just have two if statements and get rid of the `else ifs`

Reply:
if this is what youre thinking:

`    if (title.length === 0) {
      setTitleError('error');
    } else {
      productRequests.push(newFeedback);
      navigate('/');
    }

    if (feedback.length === 0) {
      setFeedbackError('error');
    } else {
      productRequests.push(newFeedback);
      navigate('/');
    }`

If one is missing it will still add the feedback and not trigger the validation.

Reply:
I was thinking this:

```
type FormErrors = Record<string, string>;
const [formErrors, setFormErrors] = useState<Record<FormErrors>({});
....
const submitFeedback = (event: any): void => {
    const newId = productRequests.length + 1;
    const newFeedback = {
      id: newId,
      title,
      category,
      upvotes: 0,
      status: 'suggestion',
      description: feedback,
      comments: [],
    };
    
    if (!title.length) {
      setFormErrors({...formErrors, title: "Length cannot be zero" });
    } else if(!feedback.length) {
      setFormErrors({...formErrors, feedback: "Length cannot be zero" });
    } else {
      setFormErrors({} as FormErrors);
      productRequests.push(newFeedback);
      navigate('/');
    }
}
```

Then you can conditionally render the errorMessage at the appropriate input as the key of this object is the `name` of the input.